tools/extract-changelog.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# Written by MRDGH2821 <ask.mrdgh2821@outlook.com>
+# This code is licensed under the MIT license.
+
+
+import os
+import re
+import sys
+
+
+def extract_changelog(folder_path, version):
+    changelog_file_path = os.path.join(folder_path, "CHANGELOG.md")
+    with open(changelog_file_path, "r") as f:
+        content = f.read()
+
+    # Find the version header
+    version_header = re.search(rf"^## v{version}$", content, re.MULTILINE)
+    if version_header:
+        version_header = version_header.group()
+        # Find the section corresponding to the version
+        section = re.search(
```

Comment:
Since the version-header search above is now a complete line-match, extract the body in the same regex instead of doing the search twice.

Reply:
The version number is derived from git tag, for eg:
If the git tag is `"self-building-factory-5.0.0"`,
`folder=self-building-factory` & `version=5.0.0`.

This is getting extracted from `.github/workflows/release-the-blueprint.yaml#L20` and passed by `.github/workflows/release-the-blueprint.yaml#L36`.

`CHANGELOG.md` has headers with markdown h2, i.e., `## 5.0.0`, so regex match is needed still.

You may use this following bash script to test things out.

```bash
#!/usr/bin/env bash

GITHUB_REF="self-building-factory-5.0.0"
folder=${GITHUB_REF#*/tags/}
folder=${folder%-*}
version=${GITHUB_REF##*/}
version=${version//[!0-9.]/}

# release_name=$(echo "${folder//-/ }" | sed -e 's/\b\(.\)/\u\1/g')
release_name=$(echo "${folder//-/ }" | sed -e "s/\b\(.\)/\u\1/g;s/Brians/Brian's/g")
release_name="$release_name $version"

echo "folder=$folder"
echo "version=$version"
echo "release_name=$release_name"
```

Reply:
I understand that the regex is needed but a single regex can find the desired header **and** extract the body.  You shouldn't need to do the regex match twice.